addon/pods/components/frost-list-item/component.js

```diff
@@ -12,18 +12,30 @@ export default Ember.Component.extend({
   isSelected: Ember.computed.reads('model.isSelected'),
 
   onclick: Ember.on('click', function (event) {
-    if (!Ember.ViewUtils.isSimpleClick(event)) {
+    if (!Ember.ViewUtils.isSimpleClick(event) && !event.shiftKey && !event.metaKey && !event.ctrlKey) {
       return true
     }
 
     event.preventDefault()
     event.stopPropagation()
 
     if (_.isFunction(this.get('_frostList.onSelect'))) {
-      this.get('_frostList.onSelect')({
-        record: this.get('model'),
-        isSelected: !this.get('model.isSelected')
-      })
+      let isTargetSelectionIndicator = $(event.target).hasClass('frost-list-selection-indicator') // eslint-disable-line no-undef
+      if (event.shiftKey && (!this.get('_frostList.persistedClickState.isSelected')) && !this.get('isSelected')) {
```

Comment:
I'm uncomfortable with the way we're accessing state and calling functions on a parent - I've raised an issue to look for an alternative mechanism to handle this